### PR TITLE
Port TG's vis_overlays subsystem

### PR DESCRIPTION
**Pull request by PidgeyThePirate**
## What Does This PR Do
Ports TG's vis_overlays subsystem. Also alters monitors to use the new system and adjusts their lighting slightly to suit.

**Port Links:**
Comfy computer lighting and a vis_contents overlay system foundation 
https://github.com/tgstation/tgstation/pull/39973

Better Glowing Lights
https://github.com/tgstation/tgstation/pull/49454

Profiling the relevant procs shows that the vis_overlay fire() proc appears to average 0.003 seconds per proc over 8 or so procs for 10 minutes solo, the rest stay 0.000. How it scales remains to be seen, but I don't think it'll be too bad.

## Why It's Good For The Game
It's pretty, and opens the door for some other cool visual effects, such as partially transparent items or reagents that glow in a beaker. I've just done the monitors for now though to minimize PR conflicts.

## Images of changes
image
emissiveblock

## Changelog
:cl:
add: Monitors now glow brightly in the dark. The light they emit has been dimmed slightly as a result.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/computer.dm [hunk 1]
```diff
@@ -73,19 +73,18 @@
 	flickering = FALSE
 
 /obj/machinery/computer/update_icon()
-	overlays.Cut()
-	if((stat & NOPOWER) || force_no_power_icon_state)
-		if(icon_keyboard)
-			overlays += image(icon,"[icon_keyboard]_off",overlay_layer)
+	. = ..()
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/computer.dm:76]: we don't have this system at all

Comment (PidgeyThePirate) [reply code/game/machinery/computer/computer.dm:76]: Oh. Back to the drawing board, I guess. It doesn't work properly, anyway. The glowing screens still draw on top of people no matter how I try to set it up.

File: code/__DEFINES/subsystems.dm [hunk 2]
```diff
@@ -46,6 +46,7 @@
 // The numbers just define the ordering, they are meaningless otherwise.
 #define INIT_ORDER_PROFILER	101
 #define INIT_ORDER_TITLE 100 // Load this quickly so people dont see a blank lobby screen
+#define INIT_ORDER_VIS 22
```
Comment (AffectedArc07) [inline code/__DEFINES/subsystems.dm:49]: Should this really be loading this soon? Especially before the GC.

**Comment by AffectedArc07** [comment]
Also this is just my own nitpick. The screens glowing on full brightness with no other overlays looks like a fault with shading, not an aesthetic. If you can dim the screen slightly I'll be much happier with that.

File: code/__DEFINES/subsystems.dm (hunk 2 above)
Comment (PidgeyThePirate) [reply code/__DEFINES/subsystems.dm:49]: Duly noted. I put it where there was an empty spot. Where do you think would be the best spot. After lighting, maybe?

**Comment by PidgeyThePirate** [comment]
I could see about reducing the alpha of the alpha mask a bit. I'm not totally sure how, though.

**Review by Kyep — APPROVED** [review]

File: code/_onclick/hud/plane_master.dm [hunk 3]
```diff
@@ -43,3 +43,48 @@
 /obj/screen/plane_master/lighting/backdrop(mob/mymob)
 	mymob.overlay_fullscreen("lighting_backdrop_lit", /obj/screen/fullscreen/lighting_backdrop/lit)
 	mymob.overlay_fullscreen("lighting_backdrop_unlit", /obj/screen/fullscreen/lighting_backdrop/unlit)
+
+/obj/screen/plane_master/lighting/Initialize()
+	. = ..()
+	filters += filter(type="alpha", render_source=EMISSIVE_RENDER_TARGET, flags=MASK_INVERSE)
+	filters += filter(type="alpha", render_source=EMISSIVE_UNBLOCKABLE_RENDER_TARGET, flags=MASK_INVERSE)
```
Comment (dearmochi) [inline code/_onclick/hud/plane_master.dm:50]:
```suggestion
	filters += filter(type = "alpha", render_source = EMISSIVE_RENDER_TARGET, flags = MASK_INVERSE)
	filters += filter(type = "alpha", render_source = EMISSIVE_UNBLOCKABLE_RENDER_TARGET, flags = MASK_INVERSE)
```

File: code/_onclick/hud/plane_master.dm [hunk 4]
```diff
@@ -43,3 +43,48 @@
...
+  */
+/obj/screen/plane_master/emissive
+	name = "emissive plane master"
+	plane = EMISSIVE_PLANE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	render_target = EMISSIVE_RENDER_TARGET
+
+/obj/screen/plane_master/emissive/Initialize()
+	. = ..()
+	filters += filter(type="alpha", render_source=EMISSIVE_BLOCKER_RENDER_TARGET, flags=MASK_INVERSE)
```
Comment (dearmochi) [inline code/_onclick/hud/plane_master.dm:66]:
```suggestion
	filters += filter(type = "alpha", render_source = EMISSIVE_BLOCKER_RENDER_TARGET, flags = MASK_INVERSE)
```

File: code/modules/lighting/emissive_blocker.dm [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	return
+
+/atom/movable/emissive_blocker/blob_act()
+	return
+
+/atom/movable/emissive_blocker/onTransitZ()
+	return
+
+//Prevents people from moving these after creation, because they shouldn't be.
+/atom/movable/emissive_blocker/forceMove(atom/destination, no_tp=FALSE, harderforce = FALSE)
```
Comment (dearmochi) [inline code/modules/lighting/emissive_blocker.dm:42]:
```suggestion
/atom/movable/emissive_blocker/forceMove(atom/destination, no_tp = FALSE, harderforce = FALSE)
```

**Review by dearmochi — APPROVED** [review]
Just a few nits, looks good otherwise

**Comment by AffectedArc07** [comment]
Consider also adding in https://github.com/tgstation/tgstation/pull/58130 as it was optimised to not use extra atoms

**Comment by dearmochi** [comment]
No activity, closing due to staleness.

**Pull request closed without merging**